MapboxCoreNavigation/CLLocation.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import CoreLocation
+import MapboxDirections
+
+public extension CLLocation {
+    /**
+     Returns a `CLLocation` infront of the provided location given the interval
+     with the current speed taken into account.
+     */
+    public func advance(by interval: TimeInterval) -> CLLocation {
```

Comment:
Replace `advance` with `advanced`: `advance` is an imperative form, which according to the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) means it has a side effect, whereas this method only makes and returns a copy.